hip/babeltrace_hip.c

```diff
@@ -0,0 +1,208 @@
+#include "babeltrace_hip.h"
+
+struct hip_event_callbacks * hip_create_event_callbacks(const char *name) {
+    intptr_t mem = (intptr_t)calloc(1, sizeof(struct hip_event_callbacks) + strlen(name) + 1);
+    struct hip_event_callbacks * callbacks = (struct hip_event_callbacks *)mem;
+    callbacks->name = (const char *)(mem + sizeof(struct hip_event_callbacks));
+    strcpy((char *)(callbacks->name), name);
+    utarray_new(callbacks->callbacks, &ut_ptr_icd);
+    return callbacks;
+}
+
+void hip_register_dispatcher(struct hip_dispatch *hip_dispatch, const char *name, hip_dispatcher_t *dispatcher) {
+    struct hip_event_callbacks *callbacks = NULL;
+    HASH_FIND_STR(hip_dispatch->event_callbacks, name, callbacks);
+    if (!callbacks) {
+        callbacks = hip_create_event_callbacks(name);
+        HASH_ADD_KEYPTR(hh, hip_dispatch->event_callbacks, callbacks->name, strlen(callbacks->name), callbacks);
+    }
+    callbacks->dispatcher = dispatcher;
+}
+
+void hip_register_callback(struct hip_dispatch *hip_dispatch, const char *name, void *func) {
+    struct hip_event_callbacks *callbacks;
```

Comment:
```suggestion
    struct hip_event_callbacks *callbacks = NULL ;
```

Reply:
Not required as `HASH_FIND_STR` always sets the third argument. The compiler would have complained otherwise.